src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.BeforeCommon.targets

```diff
@@ -188,4 +188,7 @@ Copyright (c) .NET Foundation. All rights reserved.
     <ProjectCapability Include="GenerateDocumentationFile" />
   </ItemGroup>
 
+  <PropertyGroup>
+    <UseAppHost Condition=" '$(SelfContained)' != 'true' and '$(PackAsTool)' == 'true' ">false</UseAppHost>
```

Comment:
This is tricky. I don't have a good solution:

This line need to be inside a target. Since PackAsTool is set in csproj.
And it need to before Microsoft.NET.Sdk.targets (basically command target), so "RunCommand" property can use the correct value.

I try to keep PackAsTool logic in one place. But I cannot in this case. So far that's where is think is the best

@peterhuene @nguerrera

Reply:
ping :)

Reply:
I think the second best place to put this would be where we decide the AppHost default: 

https://github.com/dotnet/sdk/blob/f97c9e71b981b8e328a820b51d8a5fc147f33a24/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets#L106

In the same way that ideally these targets wouldn't know about PackAsTool, you can almost argue that PackAsTool shouldn't need to know about above. We either keep PackAsTool with PackAsTool or UseAppHost with UseAppHost. And only the latter actually works. ;)

Reply:
I'd second that.  I think we can work this logic into the place we default `UseAppHost`.